packages/remirror__extension-link/src/link-extension.ts

```diff
@@ -739,6 +835,91 @@ export class LinkExtension extends MarkExtension<LinkOptions> {
 
     return autoLinkAllowedTLDs.includes(tld);
   }
+
+  private findURL(input: string): string | undefined {
+    for (const match of findMatches(input, this.options.autoLinkRegex)) {
+      const urlMatch = getMatchString(match);
+
+      if (!this.isValidUrl(urlMatch)) {
+        continue;
+      }
+
+      return urlMatch;
+    }
+
+    return;
```

Comment:
Loop might be unnecessary here - we have already split the text on whitespace characters, so there should only be one link within each sub string?

```suggestion
    const matches = findMatches(input, this.options.autoLinkRegex);
    
    if (!matches || !this.isValidUrl(match[0])) {
      return;
    }
    
    return matches[0];
```

Reply:
Absolutely was meaning to that

Reply:
Absolutely was meaning to do that